server/filestore.go

```diff
@@ -4426,6 +4426,10 @@ func (mb *msgBlock) generatePerSubjectInfo() error {
 	for seq := fseq; seq <= lseq; seq++ {
 		sm, err := mb.cacheLookup(seq)
 		if err != nil {
+			// Since we are walking by sequence we can ignore some errors that are benign to rebuilding our state.
```

Comment:
Since this seem to have been introduced by https://github.com/nats-io/nats-server/pull/2857, are the changes done there safe, or should the same type of error check be done?
https://github.com/nats-io/nats-server/pull/2857/files#diff-384c189826934c9a6fc3554dafc63dab2076245010e3d6fce5c71a93e15e9877R606
https://github.com/nats-io/nats-server/pull/2857/files#diff-384c189826934c9a6fc3554dafc63dab2076245010e3d6fce5c71a93e15e9877R4374

Reply:
Good idea, will take a closer look.

Reply:
I believe they are correct, however I have seen a report that I think is from this move so I may "undo" that one..

https://github.com/nats-io/nats-server/pull/2857/files#diff-384c189826934c9a6fc3554dafc63dab2076245010e3d6fce5c71a93e15e9877R319

Reply:
Was hoping "no message cache" thing was similar to this one but I don't think its is per se. Still investigating.

Reply:
FYI let me do the flip back on this PR as well. Will post new CL then squash after.